game/scripts/vscripts/units/swiper.lua

```diff
@@ -0,0 +1,222 @@
+	end
+
+	-- canUseRush = true
+
+	-- Swipe
+
+	local swipeRange = thisEntity.hFrontswipeAbility:GetCastRange(thisEntity:GetAbsOrigin(), thisEntity)
+
+	if thisEntity.hFrontswipeAbility:IsCooldownReady() then
+		local frontSwipeEnemies = thisEntity.hFrontswipeAbility:FindUnitsInCone(
+			thisEntity:GetAbsOrigin(),
+			thisEntity:GetForwardVector(),
+			swipeRange,
+			swipeRange*2,
+			thisEntity:GetTeamNumber(),
+			DOTA_UNIT_TARGET_TEAM_ENEMY,
+			DOTA_UNIT_TARGET_ALL,
+			DOTA_UNIT_TARGET_FLAG_NONE,
+			FIND_CLOSEST)
+
+		if #frontSwipeEnemies > 0 then
+			return CastFrontswipe()
+		end
+	end
+
+	-- Thrust
+
+	local thrustRange = thisEntity.hThrustAbility:GetSpecialValueFor("range")
+	local thrustWidth = thisEntity.hThrustAbility:GetSpecialValueFor("width")
+	local thrustMinRange = thisEntity.hThrustAbility:GetSpecialValueFor("target_min_range")
+	local thrustTarget
+	local thrustCount = 0
+
+	if thisEntity.hThrustAbility:IsCooldownReady() then
+		for k,v in pairs(enemies) do
+			local d = (v:GetAbsOrigin() - thisEntity:GetAbsOrigin()):Length2D()
+			if d < thrustRange and d > thrustMinRange then
+				local thrustEnemies = FindUnitsInLine(
+					thisEntity:GetTeamNumber(),
+					thisEntity:GetAbsOrigin(),
+					v:GetAbsOrigin(),
+					nil,
+					thrustWidth,
+					DOTA_UNIT_TARGET_TEAM_ENEMY,
+					DOTA_UNIT_TARGET_ALL,
+					DOTA_UNIT_TARGET_FLAG_NONE)
+
+				if #thrustEnemies > thrustCount then
+					thrustCount = #thrustEnemies
+					thrustTarget = v:GetAbsOrigin()
+				end
+			end
+		end
+
+		if thrustTarget then
+			return CastThrust( thrustTarget )
+		end
+	end
+
+	-- Reaper's Rush
+
+	local reapersMinRange = thisEntity.hReapersRushAbility:GetSpecialValueFor("min_range")
+	local reapersMaxRange = thisEntity.hReapersRushAbility:GetSpecialValueFor("max_range")
+	local reapersRushRadius = thisEntity.hReapersRushAbility:GetSpecialValueFor("radius")
+	local reapersRushTarget
+	local moveToTarget
+	local reapersRushCount = 0
+	for k,v in pairs(enemies) do
+		local d = (v:GetAbsOrigin() - thisEntity:GetAbsOrigin()):Length2D()
+		local closeEnemies = FindUnitsInRadius(
+			thisEntity:GetTeamNumber(),
+			v:GetOrigin(), nil,
+			reapersRushRadius,
+			DOTA_UNIT_TARGET_TEAM_ENEMY,
+			DOTA_UNIT_TARGET_HERO,
+			DOTA_UNIT_TARGET_FLAG_MAGIC_IMMUNE_ENEMIES + DOTA_UNIT_TARGET_FLAG_FOW_VISIBLE,
+			FIND_CLOSEST,
+			false)
+
+		if #closeEnemies > reapersRushCount then
+			if d < reapersMaxRange and d > reapersMinRange then
+				reapersRushCount = #closeEnemies
+				reapersRushTarget = v
+			end
+			moveToTarget = v
+		end
+	end
+
+	if thisEntity.hReapersRushAbility:IsCooldownReady() and reapersRushCount > 0 and reapersRushTarget and canUseRush then
+		return CastReapersRush( reapersRushTarget:GetAbsOrigin() )
+	end
+
+	if moveToTarget and thisEntity:IsIdle() then
+		ExecuteOrderFromTable({
+			UnitIndex = thisEntity:entindex(),
+			OrderType = DOTA_UNIT_ORDER_MOVE_TO_TARGET,
+			TargetIndex = moveToTarget:entindex()
+		})
+	end
+
+	return 0.5
```

Comment:
@l34Um1 I'm not sure if it was decided in the documentation but I think this boss should fall back to right click if all the abilities are in CD and there are enemies nearby

Reply:
No, the abilities are his rightclicks. They shouldn't have cooldowns, he just alternates between them.

Reply:
I'm unsure what you mean with your question? His abilities aren't supposed to be "not available". He only uses abilities, never anything else.

Reply:
by "not available" i mean there should be some time window after using some ability(you can regulate swipe rate by editing the cooldown basically)

Reply:
What's important to me is that he only attacks with his swipes. How you achieve that I don't care too much about.